functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package com.redhat.che.selenium.core.workspace;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Request.Builder;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.EmbeddedMaven;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Anatolii Bazko
+ */
+public class CheStarterWrapper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CheStarterWrapper.class);
+
+  private String host;
+  private String osioUrlPart;
+  private String cheStarterURL = "http://localhost:10000";
+
+  @Inject
+  public CheStarterWrapper(
+      @Named("che.osio.url") String osioUrlPart,
+      @Named("che.host") String cheHost,
+      @Named("che.chromedriver.port") String chromedriverPort
+  ) throws IOException, InterruptedException {
+    this.host = cheHost;
+    this.osioUrlPart = osioUrlPart;
+    /* RUN CHROMEDRIVER */
+    String chromeDriverCheckCommand =
+        "lsof -i TCP | grep -q 'localhost:" + chromedriverPort + " (LISTEN)'";
```

Comment:
Have you considered running tests from a docker container? If so, will it work with this code?

Reply:
We still run them on a duffy node on ci.centos.org . Not sure why that would be required.

Reply:
I see, so maybe docker is not needed. But would it be better to pass chrome driver command as a parameter?

Reply:
Why should it be better to pass the command as a parameter? I think that it will will not change and that it is more user-friendly to set chromedriverPort then whole command. Am I missing something?

Reply:
it's not a popular practice, nowadays, to execute system processes and small scripts from the inside of java code. But feel free to leave it as is if you think it is better for the project.